libraries/Microsoft.Bot.Builder.Core.Extensions/ActivityFilterMiddleware.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class ActivityFilterMiddleware : IMiddleware
+    {
+        /// <summary>
+        /// The type of Activity that the middleware should check for. e.g. ConversationUpdated or Message
+        /// </summary>
+        private readonly string _activityType;
+
+        /// <summary>
+        /// Handler to call when a matching Activity type is received
+        /// </summary>
+        private readonly ActivityFilterHandler _activityFilterHandler;
+
+        /// <summary>
+        /// Middleware to call when a matching Activity type is received
+        /// </summary>
+        private readonly IMiddleware _nextMiddleware;
+
+        public ActivityFilterMiddleware(string activityType, ActivityFilterHandler handler)
+        {
+            if (string.IsNullOrEmpty(activityType))
+                throw new ArgumentNullException(nameof(activityType));
+
+            // Activity types can be found in ActivityTypes enum
+            _activityType = activityType;
+            _activityFilterHandler = handler;
+        }
+
+        public ActivityFilterMiddleware(string activityType, IMiddleware nextMiddleware)
```

Comment:
>string activityType [](start = 40, length = 19)

My preference would be to make this more general, and just have this be a lamda which returns BOOL, so you can do your activity type filter like new ActivityFilterMiddleware((context) => context.Request?.Type == ActivityTypes.Message)